test/unitary/service/git/git-cli.test.ts

```diff
@@ -92,28 +92,41 @@ test("version", async () => {
 });
 
 test.each([
-  ["succes: destination does not exist", false],
-  ["failure: destination exists", true],
+  ["destination does not exist", false],
+  ["destination exists", true],
 ])("clone %p", async (_title: string, destExist: boolean) => {
   // Setup
+  const rmSyncSpy = jest.spyOn(fs, "rmSync");
+  const mkdirSyncSpy = jest.spyOn(fs, "mkdirSync");
   const dest = path.join(__dirname, "git-clone-test");
 
   // test for failure
   if (destExist) {
     fs.mkdirSync(dest);
     await git.clone(cwd, dest, "main");
+    
+    expect(rmSyncSpy).toBeCalledTimes(1);
+    expect(mkdirSyncSpy).toBeCalledTimes(2);
     const files = fs.readdirSync(dest);
-    expect(files.length).toBe(0);
+    expect(files.length).toBeGreaterThan(0);
   }
   // test for success
   else {
     await git.clone(cwd, dest, "main");
+    expect(rmSyncSpy).toBeCalledTimes(0);
+    expect(mkdirSyncSpy).toBeCalledTimes(0);
     expect(fs.existsSync(dest)).toBe(true);
+    const files = fs.readdirSync(dest);
+    expect(files.length).toBeGreaterThan(0);
   }
+
   // Clean up
+  rmSyncSpy.mockRestore();
+  mkdirSyncSpy.mockRestore();
   fs.rmSync(dest, { recursive: true });
 });
```

Comment:
initially I thought it was not necessary but I just tried to refactor and tbh it is a great idea, cleaner code.

> I think now we don't have to test for files.length to be greater than 0?
I would keep it because if the destination does not exist we want t oensure that we cloned something (and not just created the folder itself), wdyt?

Reply:
make sense